src/integrationTest/java/uk/gov/hmcts/reform/hmc/client/futurehearing/HearingManagementInterfaceHealthIndicatorIT.java

```diff
@@ -0,0 +1,174 @@
+
+    private static final String MESSAGE_MISSING_INVALID_PARAMS = "Missing or invalid request parameters";
+    private static final String MESSAGE_SERVER_ERROR = "Server error";
+
+    private static final String API_NAME_AD = "ActiveDirectory";
+    private static final String API_NAME_HMI = "HearingManagementInterface";
+
+    private final HearingManagementInterfaceHealthIndicator hmiHealthIndicator;
+
+    @Autowired
+    public HearingManagementInterfaceHealthIndicatorIT(HearingManagementInterfaceHealthIndicator hmiHealthIndicator) {
+        this.hmiHealthIndicator = hmiHealthIndicator;
+    }
+
+    @ParameterizedTest
+    @MethodSource("healthStatuses")
+    void healthShouldMatchHealthCheckStatus(Status healthStatus) {
+        stubHealthCheck(AUTH_TOKEN, healthStatus);
+
+        Health health = hmiHealthIndicator.health();
+
+        assertEquals(healthStatus, health.getStatus(), "Health status has unexpected value");
+    }
+
+    @ParameterizedTest
+    @MethodSource("activeDirectoryApiErrors")
+    void healthShouldBeDownForActiveDirectoryApiErrors(int responseStatusCode,
+                                                       String responseErrorDescription,
+                                                       List<Integer> responseErrorCodes,
+                                                       Map<String, Object> expectedDetails) {
+        stubFailToReturnToken(responseStatusCode, responseErrorDescription, responseErrorCodes);
+
+        Health health = hmiHealthIndicator.health();
+
+        assertHealthDown(health, expectedDetails);
+
+        removeStubFailToReturnToken();
+    }
+
+    @Test
+    void healthShouldBeDownForActiveDirectoryTimeout() {
+        // Needs to be used in conjunction with the readTimeout property set in the TestPropertySource annotation.
+        // Note that when openfeign is updated to v4.x the property name will need to be changed to
+        // spring.cloud.openfeign.client.config.future-hearing-api.readTimeout
+        stubFailToReturnTokenTimeout();
+
+        Health health = hmiHealthIndicator.health();
+
+        Map<String, Object> expectedDetails = Map.of(KEY_MESSAGE, "Connection/Read timeout",
+                                                     KEY_API_NAME, API_NAME_AD);
+        assertHealthDown(health, expectedDetails);
+
+        removeStubFailToReturnTokenTimeout();
+    }
+
+    @ParameterizedTest
+    @MethodSource("hmiApiErrors")
+    void healthShouldBeDownForHmiApiErrors(int responseStatusCode,
+                                           String responseMessage,
+                                           Map<String, Object> expectedDetails) {
+        stubHealthCheckThrowingError(AUTH_TOKEN, responseStatusCode, responseMessage);
+
+        Health health = hmiHealthIndicator.health();
+
+        assertHealthDown(health, expectedDetails);
+    }
+
+    private static Stream<Arguments> healthStatuses() {
+        return Stream.of(
```

Comment:
healthStatuses() found in HearingManagementInterfaceHealthIndicatorIT , FutureHearingRepositoryIT and HearingManagementInterfaceHealthIndicatorTest 

can this be moved to a common class?